### ISSUE TITLE
loki single binary helm chart v4.5.0 cannot be installed due to error `Cannot run Scalable targets (backend, read, write) without an object storage backend`

### ISSUE THREAD

**Issue by arikgrahl** [grafana/loki#8443]
**Describe the bug**

When installing the single binary helm chart of loki in v4.5.0 the installation fails with the error `Cannot run Scalable targets (backend, read, write) without an object storage backend.`

**To Reproduce**

Steps to reproduce the behavior: follow the [official instructions for the single binary helm chart](https://grafana.com/docs/loki/latest/installation/helm/install-monolithic/)

1. `helm repo add grafana https://grafana.github.io/helm-charts`
2. `helm repo update`
3.
  ```
  cat > values.yaml <<EOF
  loki:
    commonConfig:
      replication_factor: 1
    storage:
      type: 'filesystem'
  EOF
  ```
4. `helm template --values values.yaml loki grafana/loki --version 4.5.0`

**Expected behavior**

It is expected to perform a successful helm chart rendering and installation, respectively.
The installation fails with the error message `Cannot run Scalable targets (backend, read, write) without an object storage backend.` instead.

**Environment:**
 - Infrastructure: independent on infrastructure
 - Deployment tool: helm v3.11.0

**Screenshots, Promtail config, or terminal output**

```
$ helm repo add grafana https://grafana.github.io/helm-charts
"grafana" already exists with the same configuration, skipping
$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "grafana" chart repository
Update Complete. ⎈Happy Helming!⎈
$ cat > values.yaml <<EOF
loki:
  commonConfig:
    replication_factor: 1
  storage:
    type: 'filesystem'
EOF
$ helm template --values values.yaml loki grafana/loki --version 4.5.0
Error: execution error at (loki/templates/validate.yaml:26:4): Cannot run Scalable targets (backend, read, write) without an object storage backend.

Use --debug flag to render out invalid YAML
```

**Comment by DylanGuedes** [comment]
Hey, thanks for your report. That definitely looks like a problem.
Out of curiosity, if you uninstall everything and install it from scratch using `helm install`, does the same happen? My initial hypothesis is that you first installed without the `storage` config, which deployed Loki in Simple Scalable mode; then, you swapped to the filesystem storage, but the older pods were still there etc.

**Comment by DylanGuedes** [comment]
Forget my previous comment, here's the fix:
```yaml
loki:
  commonConfig:
    replication_factor: 1
  storage:
    type: 'filesystem'
singleBinary:
  replicas: 1
``` 
you need the `singleBinary` replicas to be higher than zero.
This is already fixed in the next docs [see here](https://grafana.com/docs/loki/next/installation/helm/install-monolithic/), you're seeing the docs that were released with Loki 2.7. Closing the issue, as the problem is fixed on the main branch.

**Comment by arikgrahl** [comment]
Thank you for your swift reply and pointing to the missing `singleBinary.replicas: 1`.
That is really helpful!
